pkg/validations/property/anyof.go

```diff
@@ -0,0 +1,175 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package property
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/crdify/pkg/config"
+	"sigs.k8s.io/crdify/pkg/validations"
+)
+
+const anyOfValidationName = "anyOf"
+
+var (
+	_ validations.Validation                                  = (*Enum)(nil)
+	_ validations.Comparator[apiextensionsv1.JSONSchemaProps] = (*Enum)(nil)
+	// ErrNilAnyOfConfig is returned when the AnyOfConfig is nil.
+	ErrNilAnyOfConfig = errors.New("AnyOfConfig must not be nil")
+	// ErrAtLeastOneSubschema is returned when an unknown AdditionPolicy is provided.
+	ErrAtLeastOneSubschema = errors.New("AnyOfConfig must contain at least one subschema")
+)
+
+// RegisterAnyOf registers the AnyOf validation
+// with the provided validation registry.
+func RegisterAnyOf(registry validations.Registry) {
+	registry.Register(anyOfValidationName, anyOfFactory)
+}
+
+// anyOfFactory is a function used to initialize an Enum validation
+// implementation based on the provided configuration.
+func anyOfFactory(cfg map[string]interface{}) (validations.Validation, error) {
+	anyOfCfg := &AnyOfConfig{}
+
+	err := ConfigToType(cfg, anyOfCfg)
+	if err != nil {
+		return nil, fmt.Errorf("parsing config: %w", err)
+	}
+
+	err = ValidateAnyOfConfig(anyOfCfg)
+	if err != nil {
+		return nil, fmt.Errorf("validating anyOf config: %w", err)
+	}
+
+	return &AnyOf{AnyOfConfig: *anyOfCfg}, nil
+}
+
+// ValidateAnyOfConfig validates the provided AnyOfConfig
+// setting default values where appropriate.
+// Currently the defaulting behavior defaults the
+// AnyOfConfig.AdditionPolicy to AdditionPolicyDisallow
+// if it is set to the empty string ("").
+func ValidateAnyOfConfig(in *AnyOfConfig) error {
+	if in == nil {
+		return fmt.Errorf("%w", ErrNilAnyOfConfig)
+	}
+
+	if len(in.Subschemas) == 0 {
+		return fmt.Errorf("%w", ErrAtLeastOneSubschema)
+	}
+
+	switch in.AdditionPolicy {
+	case AdditionPolicyAllow, AdditionPolicyDisallow:
+		// valid
+	case AdditionPolicy(""):
+		// default to disallow
+		in.AdditionPolicy = AdditionPolicyDisallow
+	default:
+		return fmt.Errorf("%w: %q", errUnknownAdditionPolicy, in.AdditionPolicy)
+	}
+
+	return nil
+}
+
+// AnyOfConfig contains additional configurations for the AnyOf validation.
+type AnyOfConfig struct {
+	// additionPolicy is how adding enums to an existing set of
+	// enums should be treated.
+	// Allowed values are Allow and Disallow.
+	// When set to Allow, adding new values to an existing set
+	// of enums will not be flagged.
+	// When set to Disallow, adding new values to an existing
+	// set of enums will be flagged.
+	// Defaults to Disallow.
```

Comment:
Can you update the comments here to properly describe how the additionPolicy is applied for schemas in the `anyOf` constraints for a given property?

This looks like it is still the text copied from the `Enum` validation implementation.